### PR TITLE
Feat: distinguish image tag for Custom Environments(#1255)

### PR DESCRIPTION
**Pull request by studioego**
distinguish image tag for Custom Environments (#1255)
If you register a custom gitlab docker url, The tag shows to get only uri path's first path name or namespace's first '/' splited string
Ex) <registry URL>/<namespace>/<project>/<image>
10.20.30.10:7055/root/test/r-base:4.1
shows custom Environemts' tag: root

However, custom image name uses the path's '/' splited last string.
So, custom image's tag used as <namespace> and <project> except for the last splited string in url path.

Ex1) 10.20.30.10:7055/root/test1/r-base:4.1 -> tag: root/test1
Ex2) 10.20.30.10:7055/root/test2/r-base:4.1 -> tag: root/test2

 #1255

### CODE REVIEW

File: src/components/backend-ai-resource-broker.ts [hunk 1]
```diff
@@ -839,8 +841,14 @@ export default class BackendAiResourceBroker extends BackendAIPage {
         prefix = '';
         kernelName = specs[1];
       } else {
```
Comment (adrysn) [inline src/components/backend-ai-resource-broker.ts]: Let's add a condition that the length of `specs` is greater than 2 for future safety: `else if (specs.length > 2)`.

File: src/components/backend-ai-resource-broker.ts [hunk 2]
```diff
@@ -839,8 +841,14 @@ export default class BackendAiResourceBroker extends BackendAIPage {
...
-        prefix = specs[1];
-        kernelName = specs[2];
+        prefix = '';
+        for (let i = 1; i < specs.length -1; i++) {
+          prefix += specs[i];
+          if (i != (specs.length-2)) {
+            prefix += '/';
+          }
+        }
```
Comment (adrysn) [inline src/components/backend-ai-resource-broker.ts]: This code can be a one-liner.
```javascript
prefix = specs.slice(1, specs.length - 1).join('/')
```

**Review by adrysn — CHANGES_REQUESTED** [review]
Minor request to refactor code.

Also, when I launched a session with `cr.backend.ai/testing/kt/dfold:2205-py38-cuda11.2`, the image name in the session list is `testing`, not `dfold` or `kt/dfold`. Let's fix this issue as well.
image

File: src/components/backend-ai-resource-broker.ts (hunk 1 above)
Comment (studioego) [reply src/components/backend-ai-resource-broker.ts]: Apply it.

File: src/components/backend-ai-resource-broker.ts (hunk 2 above)
Comment (studioego) [reply src/components/backend-ai-resource-broker.ts]: also apply it

**Comment by studioego** [comment]
> Minor request to refactor code.
> 
> Also, when I launched a session with `cr.backend.ai/testing/kt/dfold:2205-py38-cuda11.2`, the image name in the session list is `testing`, not `dfold` or `kt/dfold`. Let's fix this issue as well. image

I will check more about the session list source code.

**Comment by studioego** [comment]
image

If you launched a session with cr.backend.ai/testing/kt/dfold:2205-py38-cuda11.2,
the image name in the session list is `kt/dfold`, not shown `testing`

**Review by adrysn — APPROVED** [review]
LGTM.

**Pull request merged**